components/Newsfeed/Newsfeed.tsx

```diff
@@ -0,0 +1,276 @@
+import {
+  InstantSearch,
+  Pagination
+} from "@alexjball/react-instantsearch-hooks-web"
+import { collection, query, where, getDocs } from "firebase/firestore"
+import ErrorPage from "next/error"
+import { useCallback, useEffect, useState } from "react"
+import TypesenseInstantSearchAdapter from "typesense-instantsearch-adapter"
+import { useMediaQuery } from "usehooks-ts"
+import { useAuth } from "../auth"
+import { Col, Row, Spinner } from "../bootstrap"
+import { usePublicProfile } from "../db"
+import { firestore } from "../firebase"
+import {
+  Header,
+  HeaderTitle,
+  StyledContainer
+} from "./StyledNewsfeedComponents"
+import { AlertCard } from "components/AlertCard/AlertCard"
+import { getServerConfig } from "components/search/common"
+import { SearchContainer } from "components/search/SearchContainer"
+import { SearchErrorBoundary } from "components/search/SearchErrorBoundary"
+import { initialSortByValue } from "components/search/SortBy"
+import { currentGeneralCourt } from "functions/src/shared"
+
+type ElementProps = {
+  bodyText: string
+  court: string
+  header: string
+  id: string
+  subheader: string
+  timestamp: string
+  topicName: string
+  type: string
+}
+
+type Props = ElementProps[]
+
+const searchClient = new TypesenseInstantSearchAdapter({
+  server: getServerConfig(),
+  additionalSearchParameters: {
+    query_by: "number,title,body",
+    exclude_fields: "body"
+  }
+}).searchClient
+
+export default function Newsfeed() {
+  return (
+    <SearchErrorBoundary>
+      <InstantSearch
+        indexName={initialSortByValue}
+        initialUiState={{
+          [initialSortByValue]: {
+            refinementList: { court: [String(currentGeneralCourt)] }
+          }
+        }}
+        searchClient={searchClient}
+        routing={false}
+        // routing={useRouting()}
+      >
+        <Layout />
+      </InstantSearch>
+    </SearchErrorBoundary>
+  )
+}
+
+const Layout = () => {
+  const isMobile = useMediaQuery("(max-width: 768px)")
+
+  const { user } = useAuth()
+  const { result: profile, loading } = usePublicProfile(user?.uid)
+  const uid = user?.uid
+  const subscriptionRef = collection(
+    firestore,
+    `/users/${uid}/userNotificationFeed/`
```

Comment:
What happens when `uid` is undefined? Could this blow up?

Reply:
it leads to a lack of a profile variable on ln 125 which generates the <ErrorPage> Component.

I suppose rather than a catch all Error message, could be replaced with a "you are not / need to be logged in" message or something.